wrappers/android/tools/camera/src/main/java/com/intel/realsense/camera/ControlsDialog.java

```diff
@@ -22,6 +33,100 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
         LayoutInflater inflater = activity.getLayoutInflater();
         View fragmentView = inflater.inflate(R.layout.controls_dialog, null);
 
+        // getting all existing entries
+        final String emitterEntries[] = this.getResources().getStringArray(R.array.controls_emitter_entries);
+
+        int indexOfLastItem = emitterEntries.length - 1;
+        int indexOfCurrentEmitter = 0;
+        int indexOfCurrentHwPreset = 0;
+
+        RsContext ctx = new RsContext();
```

Comment:
There is over-use of query_devices() API which affects performance.
Why the current device is not stored in the Model class and then dispatched to the View/View-Model?

Reply:
done